src/scripts/main.js

```diff
@@ -1,3 +1,42 @@
 'use strict';
 
-// write code here
+const table = document.querySelector('table');
+const rows = [...table.rows];
+
+function toNumber(str) {
+  const num = parseFloat(str.replace(/[$,]/g, ''));
+
+  return isNaN(num) ? str.trim().toLowerCase() : num;
+}
+
+// eslint-disable-next-line no-shadow
+table.addEventListener('click', (event) => {
+  const thead = event.target.closest('th');
+
+  if (thead) {
+    const index = thead.cellIndex;
+    const newCells = [];
+
+    for (let i = 1; i < rows.length - 1; i++) {
+      const row = rows[i];
+      const cell = row.cells[index];
+
+      newCells.push(cell.cloneNode(true));
+    }
+
+    newCells.sort((a, b) => {
+      const cellA = a.textContent.trim();
+      const cellB = b.textContent.trim();
+
+      if (!isNaN(toNumber(cellA)) && !isNaN(toNumber(cellB))) {
+        return toNumber(cellA) - toNumber(cellB);
+      }
```

Comment:
The condition `!isNaN(toNumber(cellA)) && !isNaN(toNumber(cellB))` is used to check if both cells contain numeric values. However, the `toNumber` function returns the original string if it cannot convert it to a number. This means that the condition might not work as intended if `toNumber` returns a non-numeric string. Consider checking if `toNumber(cellA)` and `toNumber(cellB)` are numbers directly, or adjust the `toNumber` function to return a consistent type.